dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -224,85 +236,123 @@ public virtual Task<VectorSearchResults<TRecord>> VectorizedSearchAsync<TVector>
     #region Implementation of IVectorStoreRecordCollection<ulong, TRecord>
 
     /// <inheritdoc />
-    public virtual Task<TRecord?> GetAsync(ulong key, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
+    public async Task<TRecord?> GetAsync(ulong key, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
     {
-        return this.InternalGetAsync(key, options, cancellationToken);
+        using var connection = await this.GetConnectionAsync(cancellationToken).ConfigureAwait(false);
+        return await this.InternalGetAsync(connection, key, options, cancellationToken).ConfigureAwait(false);
     }
 
     /// <inheritdoc />
-    public virtual IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<ulong> keys, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
+    public async IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<ulong> keys, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
```

Comment:
this should fix the build

```suggestion
    public async IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<ulong> keys, GetRecordOptions? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
```

Reply:
Oh right, it's also important...!